lib/ClimaCoreTempestRemap/test/netcdf.jl

```diff
@@ -0,0 +1,172 @@
+    ClimaCoreTempestRemap.remap_weights(
+        weightfile,
+        meshfile_cc,
+        meshfile_rll,
+        meshfile_overlap;
+        in_type = "cgll",
+        in_np = Nq,
+    )
+
+    datafile_rll = joinpath(OUTPUT_DIR, "data_rll.nc")
+    ClimaCoreTempestRemap.apply_remap(
+        datafile_rll,
+        datafile_cc,
+        weightfile,
+        ["xlat", "sinlong"],
+    )
+
+    nc_rll = NCDataset(datafile_rll)
+    lats = Array(nc_rll["lat"])
+    lons = Array(nc_rll["lon"])
+    @test lats ≈ -89:2:89
+    @test lons ≈ 1:2:359
+    @test Array(nc_rll["xlat"]) ≈ ones(nlon) * lats' rtol = 0.1
+    @test nc_rll["sinlong"][:, :, 1] ≈ sind.(lons) * ones(nlat)' rtol = 0.1
+    @test nc_rll["sinlong"][:, :, 2] ≈ sind.(lons .+ 30) * ones(nlat)' rtol =
+        0.1
+    close(nc_rll)
+end
+
+@testset "write remap 3d sphere data" begin
+    # generate CC mesh
+    ne = 4
+    R = 1000.0
+    nlevels = 10
+    Nq = 5
+    hdomain = Domains.SphereDomain(R)
+    hmesh = Meshes.EquiangularCubedSphere(hdomain, ne)
+    htopology = Topologies.Topology2D(hmesh)
+    quad = Spaces.Quadratures.GLL{Nq}()
+    hspace = Spaces.SpectralElementSpace2D(htopology, quad)
+
+    vdomain = Domains.IntervalDomain(
+        Geometry.ZPoint(0.0),
+        Geometry.ZPoint(50.0);
+        boundary_tags = (:bottom, :top),
+    )
+    vmesh = Meshes.IntervalMesh(vdomain, nelems = nlevels)
+    vspace = Spaces.CenterFiniteDifferenceSpace(vmesh)
+
+    hvspace = Spaces.ExtrudedFiniteDifferenceSpace(hspace, vspace)
+
+    # write mesh
+    meshfile_cc = joinpath(OUTPUT_DIR, "mesh_cc_3d.g")
+    write_exodus(meshfile_cc, htopology)
+
+    # write data
+    datafile_cc = joinpath(OUTPUT_DIR, "data_cc.nc")
+    NCDataset(datafile_cc, "c") do nc
+        ClimaCoreTempestRemap.write_space_cgll!(nc, hvspace)
+        ClimaCoreTempestRemap.write_var_cgll!(
+            nc,
+            "xlat",
+            Fields.coordinate_field(hvspace).lat,
+        )
+        ClimaCoreTempestRemap.write_var_cgll!(
+            nc,
+            "xz",
+            Fields.coordinate_field(hvspace).z,
+        )
+    end
+
+    nlat = 90
+    nlon = 180
+    meshfile_rll = joinpath(OUTPUT_DIR, "mesh_rll.g")
+    ClimaCoreTempestRemap.rll_mesh(meshfile_rll; nlat = nlat, nlon = nlon)
+
+    meshfile_overlap = joinpath(OUTPUT_DIR, "mesh_overlap.g")
+    ClimaCoreTempestRemap.overlap_mesh(
+        meshfile_overlap,
+        meshfile_cc,
+        meshfile_rll,
+    )
+
+    weightfile = joinpath(OUTPUT_DIR, "remap_weights.nc")
+    ClimaCoreTempestRemap.remap_weights(
+        weightfile,
+        meshfile_cc,
+        meshfile_rll,
+        meshfile_overlap;
+        in_type = "cgll",
+        in_np = Nq,
+    )
+
+    datafile_rll = joinpath(OUTPUT_DIR, "data_rll_3d.nc")
+    ClimaCoreTempestRemap.apply_remap(
+        datafile_rll,
+        datafile_cc,
+        weightfile,
+        ["xlat", "xz"],
+    )
+
+    nc_rll = NCDataset(datafile_rll)
```

Comment:
use the `NcDataset() do ... end` form here to ensure that the file is closed consistently?